io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -388,6 +388,15 @@ impl Lvol {
                         })
                     }
                 },
+                SnapshotXattrs::SnapshotUuid => match params.snapshot_uuid() {
```

Comment:
Why is uuid an xattr? This means it can potentially be nil if the attrs get corrupted for some reason?
Did we consider storing uuid in the uuid of the spdk snapshot (though not sure if that one is "persistent"?) Alternatively could we set the uuid as the spdk snapshot name, and store the name as an xattr? So this way we might loose the name, but never the uuid?

Reply:
> Why is uuid an xattr? This means it can potentially be nil if the attrs get corrupted for some reason? Did we consider storing uuid in the uuid of the spdk snapshot (though not sure if that one is "persistent"?) Alternatively could we set the uuid as the spdk snapshot name, and store the name as an xattr? So this way we might loose the name, but never the uuid?

Currently the newly added SPDK API take one xattr array where 2 parameters are reserved for `name` and `uuid` of the snapshot and other parameters are dynamically controlled by dataplane. This has simplified the interface and can be used for future expansion to add new attribute. These all attributes are persistent to blobstore so chances of loosing these data might be very less. As We are considering the `uuid` from xattr and save it to snapshot lvol uuid, there is always a copy in lvol.

Reply:
Ok sounds good